### PR TITLE
Refactor the create function

### LINKED ISSUE #154: Refactor the create function
/kind cleanup

The create function is pretty hard to follow and needs to be refactored into a switch case.
```
func (r *DockerMachineReconciler) create(
	ctx context.Context,
	c *capiv1alpha2.Cluster,
	machine *capiv1alpha2.Machine,
	dockerMachine *infrastructurev1alpha2.DockerMachine,
) (ctrl.Result, error) {
```

### PR DESCRIPTION
**Pull request by ericzhang08**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #154 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by chuckha** [comment]
/approve

👍

File: controllers/dockermachine_controller.go [hunk 1]
```diff
@@ -158,75 +158,56 @@ func (r *DockerMachineReconciler) create(
...
 		return ctrl.Result{}, nil
 	}
+	return ctrl.Result{}, nil
+}
 
-	// If there are no control plane then we should hold off on joining workers
-	if len(controlPlanes) == 0 {
-		r.Log.Info("Sending machine back since there is no cluster to join", "machine", machine.Name)
-		return ctrl.Result{RequeueAfter: time.Second * 30}, nil
+func (r *DockerMachineReconciler) createFirstControlPlaneNode(
```
Comment (detiber) [inline controllers/dockermachine_controller.go]: It doesn't look like this method is used anywhere, can it be removed?

Comment (ericzhang08) [reply controllers/dockermachine_controller.go]: Of course. I have deleted it.

**Comment by chuckha** [comment]
apologies for the hasty approve, this looks really good now!
